src/python/pants/option/option_types.py

```diff
@@ -14,6 +15,11 @@
 _ValueT = TypeVar("_ValueT")
 # NB: We don't provide constraints, as our `XListOption` types act like a set of contraints
 _ListMemberType = TypeVar("_ListMemberType")
+_HelpT = Union[Callable[[Any], str], str]
+
+
+def _eval_maybe_helpt(help: _HelpT, subsystem_cls: Any) -> str:
```

Comment:
ditto `type{Subsystem]`. Even if MyPy doesn't like it, imo it would be helpful to have and use `# type: ignore` becauaes it makes the code self-documenting.